graphics/opencv4/Portfile

```diff
@@ -190,295 +163,197 @@ configure.args-append \
-                        sha256  f16aa7c7960e6a0d855b681d19437d4e05bc8abd6af5adf5ab86f37d1019d663 \
-                        size    60120338
-
-    depends_lib-append \
-                        port:protobuf3-cpp \
-                        port:google-glog \
-                        port:gflags \
-                        port:ceres-solver
-
-    post-extract {
-        move_gh_repo ${github.author}-${github.project}-* ${distname}
-        move_gh_repo ${github.author}-${github.project}_contrib-* ${github.project}_contrib-${version}
-    }
-
-    pre-patch {
-        delete ${worksrcpath}/${github.project}_contrib
-        ln -s ${workpath}/${github.project}_contrib-${version} ${worksrcpath}/${github.project}_contrib
+                    ${github.project}_contrib-${version}${extract.suffix} \
+                    rmd160  93552a948ff67151b7d1b00f60333d870d4c2e8f \
+                    sha256  2cb2f0529611b863ea87202252b47f7c11cdddb5b9d8ff9cbb1efbf19732abe6 \
+                    size    60605129
+
+    if {${name} eq ${subport}} {
+        pre-patch {
+            delete      ${worksrcpath}/${github.project}_contrib
+            ln -s ${workpath}/${github.project}_contrib-${version} ${worksrcpath}/${github.project}_contrib
+        }
     }
 
     configure.args-append \
-                        -DOPENCV_EXTRA_MODULES_PATH=${worksrcpath}/${github.project}_contrib/modules
+                    -DOPENCV_EXTRA_MODULES_PATH=${worksrcpath}/${github.project}_contrib/modules
 }
 
-variant qt4 conflicts qt5 description {Build with Qt4 Backend support} {
-    PortGroup           qt4 1.0
-    configure.args-replace \
-                        -DWITH_QT:BOOL=OFF \
-                        -DWITH_QT=4
+# create Python subports
+set python_versions {37 38}
+foreach v ${python_versions} {
+    subport py${v}-${github.project} {
+        PortGroup   python 1.0
+        python.default_version  ${v}
+    }
 }
 
-variant qt5 conflicts qt4 description {Build with Qt5 Backend support} {
-    PortGroup           qt5 1.0
-    configure.args-replace \
-                        -DWITH_QT:BOOL=OFF \
-                        -DWITH_QT=5
-}
+# Python bindings for supported Python versions
+if {[string match "py*" ${subport}]} {
+    set full_version    4.5.1.48
+    github.setup    skvark opencv-python [lindex [split ${full_version} .] end]
+    version         ${full_version}
+    categories-append   graphics
+    fetch.type      git
+    distname        ${github.project}-${full_version}
+    license         MIT
+
+    description     Automated CI toolchain to produce precompiled \
+                    opencv-python, opencv-python-headless, \
+                    opencv-contrib-python, and opencv-contrib-python-headless \
+                    packages.
+    long_description \
+                    This package provides Python ${python.branch} \
+                    bindings, modules, and scripts.
+
+    homepage        https://github.com/skvark/opencv-python
```

Comment:
I'll take a look at this project. In the interim, let's revert all of these changes to opencv4, per Frank's comment.

Reply:
Sorry for stepping all your great work! Done.

Also updated to version 4.5.1.

Reply:
No worries, the key is that the reversions weren't merged to master. But it looks like you've spent plenty of time on this PR, and I'm looking forward to trying out all of the new python modules you're adding!